packages/react-native/Libraries/LinkingIOS/RCTLinkingManager.mm

```diff
@@ -153,12 +154,15 @@ - (void)handleOpenURLNotification:(NSNotification *)notification
 
 RCT_EXPORT_METHOD(getInitialURL : (RCTPromiseResolveBlock)resolve reject : (__unused RCTPromiseRejectBlock)reject)
 {
+  RCTBridge *bridge = self.bridge;
+  if (!bridge) {
+    bridge = [RCTBridge currentBridge];
+  }
```

Comment:
Modules are always scoped to a single bridge. Accessing the singleton here is incorrect.

If `bridge` is nil, it means this module is being destroyed and the module should be torn down.

Reply:
emm,in the bridgeless mode, bridge ivar is always nil because it is a turbo module.

Reply:
Ah, I think there's a compat layer which provides an RCTBridge like component, which may not be turned on then.

`launchOptions` is not supported in it (yet) - https://github.com/facebook/react-native/blob/main/packages/react-native/React/Base/RCTBridgeProxy.mm#L192

Reply:
Hmm, isn't this part of bridgeless compat? @cipolleschi ?

Accessing the singleton here is an anti-pattern and should be avoided.

Reply:
I agree with @javache that we should not access the `[RCTBridge currentBridge]`. 

In bridgeless mode, `self.bridge` would be nil, and the `currentBridge` will return the `bridgeProxy` that behaves like a bridge. So, this technically works.
However, I'd rather fix the issue properly, looking if we can find a way to remove the reference to the bridge altogether. There are only three reference to the bridge in the `RCTLinkingManager`: if we find a place to store the `launchOptions` which can work for both Old and New Architecture, that would be great.

Reply:
Also... this should actually works already: in bridgeless mode, this module should be created by the `RCTTurboModuleManager` [here](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModuleManager.mm#L633).

This method checks whether there exists a `_bridge` or a `_bridgeProxy` and it sets the `bridge` ivar with the `_bridge` or the `_bridgeProxy` object ([here](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModuleManager.mm#L678)).

So, my guess is that the
```objc
if (!bridge) {
    bridge = [RCTBridge currentBridge];
  }
```

Is not necessary at all, but we need to store the `launchOptions` in the `bridgeProxy`.

Reply:
@cipolleschi you can see it also check `_isLegacyModuleClass` of `class`, so only legacy module can be set _bridgeProxy in bridgeless mode.
```
      if (_bridge) {
        [(id)module setValue:_bridge forKey:@"bridge"];
      } else if (_bridgeProxy && [self _isLegacyModuleClass:[module class]]) {
        [(id)module setValue:_bridgeProxy forKey:@"bridge"];
      }
```

Reply:
True, but `_isLegacyModuleClass` checks [this](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModuleManager.mm#L488-L492). I think that this should be turned on in the OSS to simplify the migration...